controllers/serviceaccount_controller.go

```diff
@@ -184,7 +182,8 @@ func (r *ServiceAccountReconciler) Reconcile(ctx context.Context, req reconcile.
 	}()
 
 	if !vsphereCluster.DeletionTimestamp.IsZero() {
-		return ctrl.Result{}, r.reconcileDelete(ctx, clusterContext)
+		controllerutil.RemoveFinalizer(clusterContext.VSphereCluster, vmwarev1.ProviderServiceAccountFinalizer)
```

Comment:
q: should we 
1. deprecate ProviderServiceAccountFinalizer (and remove it in future versions)
2. stop adding the finalizer and instead remove it as a cleanup from older versions?

Reply:
Not sure if it's worth it. We have to keep it around for a few years in any case, and maybe we'll have some cleanup logic again in the future :)

Reply:
No strong opinions, but It would be nice to avoid having a finalizer added to an object if we don't need it.

In case we keep it around, let's just add a comment on the Finalizer/when it is used about the fact that we are keeping this around for compatibility with older reasons

Reply:
Also no strong opinions. I'm fine with deprecating. We just really have to be aware that we have to keep it for a while.

If we stop removing the finalizer too early we have a problem.

Reply:
ack, lets just deprecate and add the add a comment on the Finalizer/when it is used about the fact that we are keeping this around for compatibility with older versions

Reply:
Deprecated and changed the code to always remove the finalizer. I think when deprecating, we should already start removing it. We won't update it anyway.